### DISCUSSION TITLE
Several api keys for Alchemy Provider

### DISCUSSION THREAD

**Discussion by onemonster** [wevm/wagmi#1419] [Q&A]
```
const { chains, provider } = configureChains(
  [chain.mainnet, chain.polygon],
  [alchemyProvider({ apiKey: 'yourAlchemyApiKey' })],
)
```
There are two chains here but the alchemy provider takes only one apiKey. If you use Alchemy, you will notice that you need to create an api key per chain. Is there a way to add multiple chains to one api key? Is there a way to provide more than one apiKey to the alchemyProvider?

**Comment by tmm** [comment]
See https://github.com/wagmi-dev/wagmi/discussions/974#discussioncomment-3690465
